cmd/eksctl-anywhere/cmd/generatepackage.go

```diff
@@ -0,0 +1,100 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/spf13/cobra"
+
+	"github.com/aws/eks-anywhere/pkg/curatedpackages"
+	"github.com/aws/eks-anywhere/pkg/kubeconfig"
+	"github.com/aws/eks-anywhere/pkg/validations"
+	"github.com/aws/eks-anywhere/pkg/version"
+)
+
+type generatePackageOptions struct {
+	directory   string
+	source      curatedpackages.BundleSource
+	kubeVersion string
+	registry    string
+}
+
+var gpOptions = &generatePackageOptions{}
+
+func init() {
+	generateCmd.AddCommand(generatePackageCommand)
+	generatePackageCommand.Flags().StringVarP(&gpOptions.directory, "directory", "d", "", "Directory to save generated packages")
+	generatePackageCommand.Flags().Var(&gpOptions.source, "source", "Location to find curated packages: (cluster, registry)")
+	if err := generatePackageCommand.MarkFlagRequired("source"); err != nil {
+		log.Fatalf("Error marking flag as required: %v", err)
+	}
+	generatePackageCommand.Flags().StringVar(&gpOptions.kubeVersion, "kubeversion", "", "Kubernetes Version of the cluster to be used. Format <major>.<minor>")
+	if err := generatePackageCommand.MarkFlagRequired("directory"); err != nil {
+		log.Fatalf("Error marking flag as required: %v", err)
+	}
+	generatePackageCommand.Flags().StringVar(&gpOptions.registry, "registry", "", "Used to specify an alternative registry for package generation")
+}
+
+var generatePackageCommand = &cobra.Command{
+	Use:          "packages [flags]",
+	Aliases:      []string{"package", "packages"},
+	Short:        "Generate package(s)",
+	Long:         "This command is used to generate curated packages",
+	PreRunE:      preRunPackages,
+	SilenceUsage: true,
+	RunE:         runGeneratePackages(),
+}
+
+func runGeneratePackages() func(cmd *cobra.Command, args []string) error {
```

Comment:
curious why not just `func runGeneratePackages(cmd *cobra.Command, args []string) error` and reference with `runGeneratePackages`